engine/schema/src/main/java/com/cloud/configuration/dao/ResourceCountDaoImpl.java

```diff
@@ -154,6 +175,28 @@ public boolean updateById(long id, boolean increment, long delta) {
         return update(resourceCountVO.getId(), resourceCountVO);
     }
 
+    @Override
+    public boolean updateCountByDeltaForIds(List<Long> ids, boolean increment, long delta) {
```

Comment:
would it make sense to remove increment and check for positive/negative numbers?

the higher level call to `updateResourceCountForAccount` is only done in two places so it wouldn't be a big refactor.